tests/pytests/test_numbers.py

```diff
@@ -389,3 +389,91 @@ def testNegativeValues(env):
+    env.assertEqual(conn.execute_command('HSET', 'doc07', 'n', '1.5e-2'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc08', 'n', 'INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc09', 'n', 'inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc10', 'n', 'iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc11', 'n', '+INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc12', 'n', '+inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc13', 'n', '+iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc14', 'n', '-INF'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc15', 'n', '-inf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc16', 'n', '-iNf'), 1)
+    env.assertEqual(conn.execute_command('HSET', 'doc17', 'n', '-1'), 1)
+
+    # Test unsigned numbers
+    expected = [3, 'doc01', 'doc02', 'doc03']
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[1 1]', 'NOCONTENT')
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[1e0 1]', 'NOCONTENT')
+    env.assertEqual(res, expected)
+
+    # Test signed numbers
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[+1e0 +1]', 'NOCONTENT')
+    env.assertEqual(res, expected)
+
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[-1e0 -1]', 'NOCONTENT')
+    env.assertEqual(res, [1, 'doc17'])
+
+    # Test +inf
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:[1e6 inf]', 'NOCONTENT')
+    expected = [6, 'doc08', 'doc09', 'doc10', 'doc11', 'doc12', 'doc13']
+    env.assertEqual(res1, expected)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:[1e6 INF]', 'NOCONTENT')
+    env.assertEqual(res2, expected)
+
+    # Test -inf
+    res1 = env.cmd('FT.SEARCH', 'idx', '@n:[-inf 0]', 'NOCONTENT')
+    expected = [4, 'doc14', 'doc15', 'doc16', 'doc17']
+    env.assertEqual(res1, expected)
+    res2 = env.cmd('FT.SEARCH', 'idx', '@n:[-INF 0]', 'NOCONTENT')
+    env.assertEqual(res2, expected)
+
+    # Multiple parenthesis are allowed
+    expected = [1, 'doc05']
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[1e2 ((300]', 'NOCONTENT')
+    env.assertEqual(res, expected)
+    res = env.cmd('FT.SEARCH', 'idx', '@n:[((1 300]', 'NOCONTENT')
+    env.assertEqual(res, expected)
+
+    # invalid syntax - multiple signs are not allowed
+    env.expect('FT.SEARCH', 'idx', '@n:[--1e0 -+1]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[++1e0 +-1]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[++inf 1]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[-+inf 1]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[+-inf 1]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[--inf 1]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[1 ++inf]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[1 -+inf]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[1 +-inf]').error()
+    env.expect('FT.SEARCH', 'idx', '@n:[1 --inf]').error()
+
+def testExplainNumericRange(env):
```

Comment:
Please add comments throughout the test, or an initial comment explaining the test.

Reply:
The tests done by `testExplainNumericRange(env)` were moved to `test:testExplain()`. [a448913](https://github.com/RediSearch/RediSearch/pull/4505/commits/a448913e7960c75d10bd6c2fcf6d8ce6e519dad7)